Add check for actions triggered by cron

aramattamara opened the pull request:
Improves #5639 

### What changes did you make?
  - Added check for actions triggered by cron. 

### Why did you make the changes (we will use this info to test)?
  - #5639

aramattamara committed:
Add check for actions triggered by cron

t-will-gillis reviewed (changes requested):
Hi @aramattamara  Thank you for the PR which appears to be a failsafe to prevent a GHA from running inadvertently when a dev has authorized a token in their personal repo with the same name as the HfLA token. I agree with you that this is a worthwhile and easy ‘safety’ measure.  

For the `schedule-monthly-PREV.yml` file, ideally this will be deleted by PR #5467 before it is scheduled to run again on Nov 1st, but just in case it isn’t, this is an easy solution. Incidentally, to be doubly sure lines 4 & 5 can be replaced with `workflow_dispatch:`. (Of course, I should have done this from the beginning but I did not think the last PR was going to be active this long.)  

For the `schedule-fri-0700.yml`, I don’t believe that the repo check is needed because the `secrets.IN_PROGRESS_COLUMN_ID` only refers to HfLA’s Project Board. Since the personal repos do not know this secret, they won’t trigger the action. For that reason, and since we have not tested to confirm that there are no unexpected results from adding the repo check, I do not believe we should update this GHA (If it ain’t broke…)  

For the future, this will be a good reminder to make sure that scopes are set appropriately to  prevent individuals from accidentally triggering GHAs- thank you for pointing this out.

If you could make the change(s) to `schedule-monthly-PREV.yml`, I can re-review and merge it. 

I will add labels, and if you think of any notes to add to the description for clarification, please do.

Thanks!

aramattamara committed:
Replaced 4 and 5 line with "workflow_dispatch"

aramattamara committed:
Revert 'schedule-fri-0700.yml'

aramattamara commented:
Hi @t-will-gillis, thanks for your detailed feedback! I've implemented the changes you suggested in the `schedule-monthly-PREV.yml`. I've replaced lines 4 and 5 with `workflow_dispatch`.

I also reverted the `schedule-fri-0700.yml`, thanks.

t-will-gillis reviewed (approved):
Hey @aramattamara Thanks for the changes!
